src/mjolnir/graphbuilder.cc

```diff
@@ -200,27 +201,40 @@ void ConstructEdges(const OSMData& osmdata,
           prev_edge.attributes.way_prior = true;
         }
 
-        if (!edge.attributes.way_begin)
+        // We should add the previous edge now that we know its done
+        if (prev_edge.is_valid())
```

Comment:
again i tried to make this more clear. if we have a `prev_edge` finished and waiting to get pushed on then we do so now because we are about to overwrite it. ie queue the next one